src/controller/CHIPDeviceController.cpp

```diff
@@ -3567,17 +3573,34 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
             return;
         }
 
-        CHIP_ERROR err = ValidateJFAdminNOC(params.GetJFAdminNOC().Value());
+        P256PublicKeySpan jfAdminICACPKSpan;
+        CHIP_ERROR err = CHIP_NO_ERROR;
+
+        /* check the peer Admin NOC */
+        err = ValidateJFAdminNOC(params.GetJFAdminNOC().Value());
 
         if (err != CHIP_NO_ERROR)
         {
             ChipLogError(Controller, "JCM: Cannot validate JFAdminNOC");
+            CommissioningStageComplete(err);
+            return;
         }
 
-        CommissioningStageComplete(err);
+        /* extract and save the public key of the peer Admin ICAC */
+        err = ExtractPublicKeyFromChipCert(params.GetJFAdminICAC().Value(), jfAdminICACPKSpan);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(Controller, "JCM: Error parsing JFAdminICAC Public Key");
+            CommissioningStageComplete(err);
+            return;
+        }
 
-        break;
+        Crypto::P256PublicKey jfAdminICACPK{ jfAdminICACPKSpan };
+        mTrustedIcacPublicKeyB.SetValue(jfAdminICACPK);
```

Comment:
Does doing:
```suggestion
        mTrustedIcacPublicKeyB.Emplace(jfAdminICACPKSpan);
```
work?

Reply:
applied this suggestion.